Microsoft.Toolkit.Uwp.UI.Controls/AdaptiveGridView/AdaptiveGridView.cs

```diff
@@ -82,14 +83,23 @@ private void RecalculateLayout(double containerWidth)
             double desiredWidth = double.IsNaN(DesiredWidth) ? containerWidth : DesiredWidth;
 
             _columns = CalculateColumns(containerWidth, desiredWidth);
+            bool fillsAtLeastOneRow = true;
```

Comment:
Rather than checking if there's only one row, enforce the column count by checking your property:
```
    if ((MaintainAspectRatio == false) && Items != null && Items.Count > 0 && Items.Count < _columns)
    {
        _columns = Items.Count;
    }
```